src/lib.rs

```diff
@@ -1118,7 +1119,10 @@ where
     fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) -> Self {
         // safety: we have &mut self, so not concurrently accessed by anyone else
         let guard = unsafe { crossbeam::epoch::unprotected() };
-        let output: FlurryHashMap<K, V, RandomState> = FlurryHashMap::new();
+        let it = iter.iter();
+        let size_hint = it.size_hint().0;
```

Comment:
This won't actually work, since `with_capacity` panics if the requested capacity is 0, and the `min` of the `size_hint` could totally be `0`. I think maybe what we want is something like [what the standard library does for `Vec::from_iter`](https://doc.rust-lang.org/src/alloc/vec.rs.html#1982).

Reply:
Fair enought, but wouldn't it be better if `FlurryHashMap::with_capacity(0)` didn't panick (return `FlurryHashMap::new()` instead)?

Reply:
Quite possibly -- feel free to implement that if you want to :)

Reply:
I do like what they do in `std` for `Vec` as linked above of also calling `next()` once on the iterator to handle the special case where iterator immediately yields `None`. It's pretty smart I think.

Reply:
I didn't quite get why they did that. Could you explain a bit? I made it so that `with_capacity(0)` doesn't panick anyaway.

I'm trying to imitate what they did in `Vec::from_iter` right now.

Reply:
They did it so that if you do `from_iter` from an iterator that doesn't yield any elements, you do not allocate anything, regardless of what the `size_hint` says. At least that's my read of it.